test/test_routes.py

```diff
@@ -62,20 +62,38 @@ def test_submit_workflow(client):
     ) as mock_log:
         mock_auth.return_value = True
         mock_engine.return_value = "workflow_123"
-        response = client.post(
-            "/submit",
-            data=json.dumps(data),
-            headers={
-                "Content-Type": "application/json",
-                "Authorization": "bearer 1234",
-            },
-        )
-        assert response.status_code == 200
-        assert response.content.decode("utf-8") == '"workflow_123"'
-        mock_auth.assert_called_with(
-            token="bearer 1234", team_project="dummy-team-project"
-        )
-        mock_log.assert_called_with("check_auth_and_team_project")
+        with patch("requests.get") as mock_request:
```

Comment:
maybe it is best to move this to a separate new test? The previous version of the code seems to be meant for (also) testing the FastAPI integration. By mocking request here, I think that part is being skipped, which means your test is replacing the previous test, not adding to it. Therefore, I think it is best to move your test, with the mocked requests.get to a new separate test.